shared/actions/fs/platform-specific.desktop.js

```diff
@@ -316,38 +315,37 @@ const openAndUpload = (state, action) =>
   )
 
 const loadUserFileEdits = (state, action) =>
-  RPCTypes.SimpleFSSimpleFSUserEditHistoryRpcPromise().then(writerEdits => {
-    const tlfUpdates = Constants.userTlfHistoryRPCToState(writerEdits || [])
-    const updateSet = tlfUpdates
-      .reduce(
-        (acc: I.Set<Types.Path>, u) =>
-          Types.getPathElements(u.path).reduce((acc, e, i, a) => {
-            if (i < 2) return acc
-            const path = Types.getPathFromElements(a.slice(0, i + 1))
-            return acc.add(path)
-          }, acc),
-        I.Set()
-      )
-      .toArray()
-    // TODO (songgao): make a new action that accepts an array of updates,
-    // so that we only need to trigger one update through store/rpc/widget
-    // for all these each time.
-    return [
-      ...updateSet.map(path =>
-        FsGen.createFilePreviewLoad({
-          identifyBehavior: RPCTypes.tlfKeysTLFIdentifyBehavior.chatGui,
-          path,
-        })
-      ),
-      FsGen.createUserFileEditsLoaded({tlfUpdates}),
-    ]
-  })
+  RPCTypes.SimpleFSSimpleFSUserEditHistoryRpcPromise().then(writerEdits =>
+    FsGen.createUserFileEditsLoaded({
+      tlfUpdates: Constants.userTlfHistoryRPCToState(writerEdits || []),
+    })
+  )
 
 const openFilesFromWidget = (state, {payload: {path, type}}) => [
   ConfigGen.createShowMain(),
   ...(path ? [FsGen.createOpenPathInFilesTab({path})] : [RouteTreeGen.createSwitchTo({path: [Tabs.fsTab]})]),
 ]
 
+let pingKbfsDaemonLoopRunning = false
+function* pingKbfsDaemonLoop() {
+  if (pingKbfsDaemonLoopRunning) {
+    return
+  }
+  while (true) {
```

Comment:
Why don't we ever return from this loop?

Reply:
What I have in mind is we'd continue monitoring KBFS connections here even after KBFS is connected. It's possible to restart KBFS without restarting the whole app (e.g. enabling fuse ext).

Reply:
@jzila does it sound reasonable?

Reply:
I'm a little worried about such a loop. @chrisnojima is this a reasonable approach or should we figure out something different?

Reply:
we don't do this kind of polling a lot. is it really necessary

Reply:
Should we have a universal service->GUI RPC concerning connection status notifications?

Reply:
That'd be useful. I guess until then we'll just poll at start, and stop polling when KBFS is connected?

Reply:
Sounds good to me. But what happens if KBFS never connects?

Reply:
that'll just keep polling. We do expect KBFS to connect eventually right? Or is KBFS-less client something we want to support?

Reply:
Not to actively support it, I'm just trying to think of bugs when someone's KBFS has crashed or whatever.

Reply:
If KBFS crashes at some point after connected, we'd think we're still connected because the polling has stopped (as of last change). If KBFS crashes at start, that'd be a rare scenario where we likely have other problems too. This RPC is relatively cheap, so I think it's not a big deal. In most cases we'll detect we're connected and be done with it.